src/utils/components/HorizontalNavbar/utils/useDynamicPages.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+import { VirtualMachineModel } from 'src/views/dashboard-extensions/utils';
+
+import {
+  HorizontalNavTab,
+  isHorizontalNavTab,
+  useResolvedExtensions,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+const useDynamicPages = () => {
```

Comment:
please add return type

Reply:
although the name is useDynamicPages, it's very specific for VMs, let's rename it or add an argument(VirtualMachineModel) that this hook will filter by, and let's make this hook more generic